projects/packages/my-jetpack/_inc/hooks/use-connection-watcher/index.js

```diff
@@ -13,13 +12,16 @@ import useMyJetpackNavigate from '../use-my-jetpack-navigate';
  * the hook dispatches an action to populate the global notice.
  */
 export default function useConnectionWatcher() {
+	const [ notice, setNotice ] = useState( {
+		message: '',
+		options: {},
+	} );
```

Comment:
I wouldn't mind moving that state into `useNotice` hook, and exposing `updateNotice` function from there. I'm leaving this up to you though!